editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -2012,43 +2032,35 @@ void Node3DEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
-			if (nav_scheme == NAVIGATION_MAYA && m->is_alt_pressed()) {
+			if (zoom_mouse_preference == NAVIGATION_RIGHT_MOUSE && zoom_mod_pressed) {
 				nav_mode = NAVIGATION_ZOOM;
+			} else if (pan_mouse_preference == NAVIGATION_RIGHT_MOUSE && pan_mod_pressed) {
+				nav_mode = NAVIGATION_PAN;
+			} else if (orbit_mouse_preference == NAVIGATION_RIGHT_MOUSE && orbit_mod_pressed) {
+				nav_mode = NAVIGATION_ORBIT;
 			} else if (freelook_active) {
 				nav_mode = NAVIGATION_LOOK;
 			} else if (orthogonal) {
 				nav_mode = NAVIGATION_PAN;
 			}
 
 		} else if (m->get_button_mask().has_flag(MouseButtonMask::MIDDLE)) {
-			const Key mod = _get_key_modifier(m);
-			if (nav_scheme == NAVIGATION_GODOT) {
-				if (mod == _get_key_modifier_setting("editors/3d/navigation/pan_modifier")) {
-					nav_mode = NAVIGATION_PAN;
-				} else if (mod == _get_key_modifier_setting("editors/3d/navigation/zoom_modifier")) {
-					nav_mode = NAVIGATION_ZOOM;
-				} else if (mod == Key::ALT || mod == _get_key_modifier_setting("editors/3d/navigation/orbit_modifier")) {
-					// Always allow Alt as a modifier to better support graphic tablets.
-					nav_mode = NAVIGATION_ORBIT;
-				}
-			} else if (nav_scheme == NAVIGATION_MAYA) {
-				if (mod == _get_key_modifier_setting("editors/3d/navigation/pan_modifier")) {
-					nav_mode = NAVIGATION_PAN;
-				}
+			if (zoom_mouse_preference == NAVIGATION_MIDDLE_MOUSE && zoom_mod_pressed) {
+				nav_mode = NAVIGATION_ZOOM;
+			} else if (pan_mouse_preference == NAVIGATION_MIDDLE_MOUSE && pan_mod_pressed) {
+				nav_mode = NAVIGATION_PAN;
+			} else if (orbit_mouse_preference == NAVIGATION_MIDDLE_MOUSE && orbit_mod_pressed) {
+				nav_mode = NAVIGATION_ORBIT;
 			}
+
 		} else if (EDITOR_GET("editors/3d/navigation/emulate_3_button_mouse")) {
 			// Handle trackpad (no external mouse) use case
-			const Key mod = _get_key_modifier(m);
-
-			if (mod != Key::NONE) {
-				if (mod == _get_key_modifier_setting("editors/3d/navigation/pan_modifier")) {
-					nav_mode = NAVIGATION_PAN;
-				} else if (mod == _get_key_modifier_setting("editors/3d/navigation/zoom_modifier")) {
-					nav_mode = NAVIGATION_ZOOM;
-				} else if (mod == Key::ALT || mod == _get_key_modifier_setting("editors/3d/navigation/orbit_modifier")) {
-					// Always allow Alt as a modifier to better support graphic tablets.
-					nav_mode = NAVIGATION_ORBIT;
-				}
+			if ((!_is_shortcut_empty("spatial_editor/viewport_zoom_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_zoom_modifier_2")) && zoom_mod_pressed) {
+				nav_mode = NAVIGATION_ZOOM;
+			} else if ((!_is_shortcut_empty("spatial_editor/viewport_pan_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_pan_modifier_2")) && pan_mod_pressed) {
+				nav_mode = NAVIGATION_PAN;
+			} else if ((!_is_shortcut_empty("spatial_editor/viewport_orbit_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_orbit_modifier_2")) && orbit_mod_pressed) {
```

Comment:
```suggestion
			if (zoom_mod_pressed && (!_is_shortcut_empty("spatial_editor/viewport_zoom_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_zoom_modifier_2"))) {
				nav_mode = NAVIGATION_ZOOM;
			} else if (pan_mod_pressed && (!_is_shortcut_empty("spatial_editor/viewport_pan_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_pan_modifier_2"))) {
				nav_mode = NAVIGATION_PAN;
			} else if (orbit_mod_pressed && (!_is_shortcut_empty("spatial_editor/viewport_orbit_modifier_1") || !_is_shortcut_empty("spatial_editor/viewport_orbit_modifier_2"))) {
```
Cheaper condition first